pkg/cache/application.go

```diff
@@ -506,13 +507,33 @@ func (app *Application) onReservationStateChange(event *fsm.Event) {
 func (app *Application) handleRejectApplicationEvent(event *fsm.Event) {
 	log.Logger().Info("app is rejected by scheduler", zap.String("appID", app.applicationID))
 	// for rejected apps, we directly move them to failed state
-	dispatcher.Dispatch(NewFailApplicationEvent(app.applicationID))
+	dispatcher.Dispatch(NewFailApplicationEvent(app.applicationID,
+		fmt.Sprintf("application %s is rejected by scheduler", app.applicationID)))
 }
 
 func (app *Application) handleCompleteApplicationEvent(event *fsm.Event) {
 	// TODO app lifecycle updates
 }
 
+func (app *Application) handleFailApplicationEvent(event *fsm.Event) {
+	eventArgs := make([]string, 1)
+	if err := events.GetEventArgsAsStrings(eventArgs, event.Args); err != nil {
+		log.Logger().Error("fail to parse event arg", zap.Error(err))
+		return
+	}
+	errMess := eventArgs[0]
+	// unallocated task states include New, Pending and Scheduling
+	unalloc := app.getTasks(events.States().Task.New)
+	unalloc = append(unalloc, app.getTasks(events.States().Task.Pending)...)
+	unalloc = append(unalloc, app.getTasks(events.States().Task.Scheduling)...)
+	// publish pod level event to unallocated pods
+	for _, task := range unalloc {
+		events.GetRecorder().Eventf(task.GetTaskPod(), v1.EventTypeWarning, "ApplicationFailed",
+			"Application %s scheduling failed, reason: %s", app.applicationID, errMess)
+	}
+	log.Logger().Warn("failed to schedule app", zap.String("appID", app.applicationID))
```

Comment:
I think we have already logged this, pls remove this line.

Reply:
OK.